docker-compose.yml

```
@@ -0,0 +1,112 @@
+version: '3.7'
+services:
+    mysql:
+        image: mariadb
+        ports:
+            - '3306'
+        environment:
+            MYSQL_ROOT_PASSWORD: password
+            MYSQL_DATABASE: wordpress
+        volumes:
+            - 'mysql:/var/lib/mysql'
+    tests-mysql:
+        image: mariadb
+        ports:
+            - '3306'
+        environment:
+            MYSQL_ROOT_PASSWORD: password
+            MYSQL_DATABASE: tests-wordpress
+        volumes:
+            - 'mysql-test:/var/lib/mysql'
+    wordpress:
+        build: .
+        depends_on:
+            - mysql
+        image: wordpress
+        ports:
+            - '${WP_ENV_PORT:-8888}:80'
+        environment:
+            WORDPRESS_DB_USER: root
+            WORDPRESS_DB_PASSWORD: password
+            WORDPRESS_DB_NAME: wordpress
+        volumes: &ref_0
+            - >-
+                /Users/ockham/.wp-env/f225b6dcecc0956ba67033945142a55c/WordPress:/var/www/html
+            - >-
+                /Users/ockham/src/gutenberg:/var/www/html/wp-content/plugins/gutenberg
```

The directory of the plugin or theme we're developing -- thus, that plugin's/theme's concern.

Since [we don't need an absolute path here](https://github.com/fabric8io/docker-maven-plugin/pull/848), this can be arguably be moved into a `.docker-compose.override.json` at the root level of that plugin or theme.

```suggestion
                .:/var/www/html/wp-content/plugins/gutenberg
```

---

This can include more than just the current plugin -- it could be any plugin or theme you wish to include in the environment as well. Things like .zip or GitHub sources will be mapped in the same way.

---

I'm covering those other cases by https://github.com/WordPress/gutenberg/pull/30974#discussion_r616095095 I think?

---

I commented too quickly :)